### PR TITLE
Consistently provide all scripts in script/

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Review by Martchus — CHANGES_REQUESTED** [review]
You still need to adapt paths used to reference those scripts.

**Comment by okurz** [comment]
https://github.com/os-autoinst/os-autoinst/actions/runs/9971827632/job/27553725203?pr=1978 fails with

```
Can't locate log.pm in @INC
```

don't know why

File: script/isotovideo [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+# Details: https://progress.opensuse.org/issues/90371
+use JSON::PP;
+
+my $installprefix;    # $bmwqemu::topdir
+
+BEGIN {
+    # the following line is modified during make install
+    $installprefix = undef;
+
+    my ($wd) = $0 =~ m-(.*)/-;
```
Comment (perlpunk) [inline script/isotovideo:100]: This needs adjustment now.
```
perl script/worker --isotovideo=../os-autoinst/isotovideo --verbose
# $0 == /home/tina/openqa-toolbox/repos/os-autoinst/script/isotovideo
Can't locate log.pm in @INC (you may need to install the log module) (@INC contains: /home/tina/openqa-toolbox/repos/os-autoinst/script ...
```

Comment (perlpunk) [inline script/isotovideo:100]:
```suggestion
    use FindBin '$RealBin';
    use Cwd qw(realpath);
    my ($wd) = realpath "$RealBin/..";
```

Comment (okurz) [reply script/isotovideo:100]: I strongly suggest we keep any real content changes to the scripts outside this PR. I really only moved isotovideo to script/isotovideo here. I don't understand what `my ($wd) = $0 =~ m-(.*)/-;` is doing. Could you please create your own PR for that change, we apply that first and I will rebase my PR on top of your changes?

Comment (perlpunk) [reply script/isotovideo:100]: yep

Comment (perlpunk) [reply script/isotovideo:100]: #2518 2518

Comment (perlpunk) [reply script/isotovideo:100]: You still need to add the `/..` in your PR though

Comment (okurz) [reply script/isotovideo]: done

File: CMakeLists.txt [hunk 2]
```diff
@@ -242,10 +242,10 @@ install(
     COMPONENT "openvswitch"
     EXCLUDE_FROM_ALL
 )
-foreach (EXTRA_SCRIPT check_qemu_oom dewebsockify vnctest)
+foreach (EXTRA_SCRIPT script/check_qemu_oom script/dewebsockify script/vnctest)
     install(
         FILES "${EXTRA_SCRIPT}"
-        DESTINATION "${OS_AUTOINST_DATA_DIR}"
+        DESTINATION "${OS_AUTOINST_DATA_DIR}/script"
```
Comment (Martchus) [inline CMakeLists.txt:248]: You created a symlink for `check_qemu_oom`. This symlink should probably be installed here as well.

File: script/isotovideo [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+use FindBin '$RealBin';
+
+my $installprefix;    # $bmwqemu::topdir
+
+BEGIN {
+    # the following line is modified during make install
+    $installprefix = undef;
+
+    my $topdir = "$RealBin/.." || '..';
```
Comment (perlpunk) [inline script/isotovideo:102]: the `|| ...` is actually useless, as the lefthand side will always be true

```suggestion
    my $topdir = "$RealBin/..";
```

File: CMakeLists.txt (hunk 2 above)
Comment (okurz) [reply CMakeLists.txt:248]: I wanted the symlinks only for convenience for developers. Everything in OS_AUTOINST_DATA_DIR isn't supposed to be called by normal users and code that relies on `check_qemu_oom` calls it from the script sub directory. Hence I would not install the symlink, fine?

**Comment by b10n1k** [comment]
I dont understand the motivation of this commit. why move them in script and not in bin? which is empty btw

**Comment by okurz** [comment]
> I dont understand the motivation of this commit. why move them in script and not in bin? which is empty btw

Good question. I don't even clearly remember anymore. Let's say because https://github.com/os-autoinst/openQA/tree/master/script and https://github.com/mojolicious/mojo/tree/main/script do it like this.

**Comment by Martchus** [comment]
`bin` is a relict from the time the project was using autotools. We should remove this directory as we never want to put binaries under version control.

**Review by Martchus — APPROVED** [review]
Let's see what this breaks. (I added the "not-ready" label so you can merge this in a time window where you can follow-up on it if necessary.)

**Comment by okurz** [comment]
fixup: #2527

**Pull request merged**